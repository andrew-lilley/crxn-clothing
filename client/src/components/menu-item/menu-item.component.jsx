import React from 'react';
import { withRouter } from 'react-router-dom';
import { AssetLocation } from '../../utils';
import {
  MenuItemContainer,
  BackgroundImageContainer,
  ContentContainer,
  ContentTitle,
  ContentSubtitle
} from './menu-item.styles';

export const MenuItem = ({ title, imageUrl, size, linkUrl, match, history }) => (

  <MenuItemContainer
    size={size}
    onClick={() => history.push(`${match.url}${linkUrl}`)}>
    <BackgroundImageContainer style={{
        backgroundImage: `url(${AssetLocation(imageUrl)})`
      }} 
    />
    <ContentContainer>
      <ContentTitle>{title.toUpperCase()}</ContentTitle>
      <ContentSubtitle>SHOP NOW</ContentSubtitle>
    </ContentContainer>
  </MenuItemContainer>
);

export default withRouter(MenuItem);