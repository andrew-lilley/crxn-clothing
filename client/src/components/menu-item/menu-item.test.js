import React from 'react';
import { shallow } from 'enzyme';
import { MenuItem } from './menu-item.component';

describe('MenuItem component', () => {
  let wrapper;
  let mockMatch;
  let mockHistory;
  const linkUrl = '/hats';
  const size = 'large';
  const imageUrl = 'images/shop-img/hats/brown-brim.png';

  beforeEach(() => {
    mockMatch = {
      url: '/shop'
    };

    mockHistory = {
      push: jest.fn()
    };

    const mockProps = {
      match: mockMatch,
      history: mockHistory,
      linkUrl,
      size,
      title: 'hats',
      imageUrl
    };

    wrapper = shallow(<MenuItem { ...mockProps } />);
  });

  test('should render MenuItem component', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('should call history.push with the right string when MenuItemContainer clicked', () => {
    wrapper.find('MenuItemContainer').simulate('click');

    expect(mockHistory.push).toHaveBeenCalledWith(`${mockMatch.url}${linkUrl}`);
  });

  test('should pass size to MenuItemContainer as the prop size', () => {
    expect(wrapper.find('MenuItemContainer').prop('size')).toBe(size);
  });

});