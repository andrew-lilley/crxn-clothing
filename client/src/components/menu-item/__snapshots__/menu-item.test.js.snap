// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuItem component should render MenuItem component 1`] = `
<MenuItemContainer
  onClick={[Function]}
  size="large"
>
  <BackgroundImageContainer
    style={
      Object {
        "backgroundImage": "url(brown-brim.png)",
      }
    }
  />
  <styled.div>
    <styled.span>
      HATS
    </styled.span>
    <styled.span>
      SHOP NOW
    </styled.span>
  </styled.div>
</MenuItemContainer>
`;
